model/QtiTestsDeletedListener.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2023 (original work) Open Assessment Technologies SA.
+ */
+
+namespace oat\taoMediaManager\model;
+
+use oat\generis\model\data\Ontology;
+use oat\tao\model\media\TaoMediaException;
+use oat\tao\model\media\TaoMediaResolver;
+use oat\tao\model\resources\Exception\ClassDeletionException;
+use oat\tao\model\resources\Exception\PartialClassDeletionException;
+use oat\tao\model\resources\Service\ClassDeleter;
+use oat\taoMediaManager\model\Specification\MediaClassSpecification;
+use oat\taoQtiTest\models\event\QtiTestsDeletedEvent;
+use core_kernel_classes_Class;
+use core_kernel_classes_Resource;
+use tao_helpers_Uri;
+use Psr\Log\LoggerInterface;
+use Exception;
+use Throwable;
+
+class QtiTestsDeletedListener
+{
+    private LoggerInterface $logger;
+    private Ontology $ontology;
+    private ClassDeleter $classDeleter;
+    private TaoMediaResolver $taoMediaResolver;
+    private MediaClassSpecification $mediaClassSpecification;
+    private MediaService $mediaService;
+
+    public function __construct(
+        LoggerInterface $logger,
+        MediaService $mediaService,
+        MediaClassSpecification $mediaClassSpecification,
+        Ontology $ontology,
+        ClassDeleter $classDeleter,
+        TaoMediaResolver $taoMediaResolver
+    ) {
+        $this->logger = $logger;
+        $this->mediaService = $mediaService;
+        $this->mediaClassSpecification = $mediaClassSpecification;
+        $this->ontology = $ontology;
+        $this->classDeleter = $classDeleter;
+        $this->taoMediaResolver = $taoMediaResolver;
+    }
+
+    /**
+     * @throws PartialClassDeletionException
+     * @throws ClassDeletionException
+     * @throws Exception
+     */
+    public function handle(QtiTestsDeletedEvent $event): void
+    {
+        $assetIds = [];
+
+        foreach (array_unique($event->getReferencedResources()) as $ref) {
+            try {
+                $id = $this->taoMediaResolver->resolve($ref)->getMediaIdentifier();
+
+                if (!isset($assetIds[$id])) {
+                    $assetIds[$id] = $id;
+
+                    $this->deleteAsset($id);
+                }
```

Comment:
I think here is the place you could add a log (the only one apart of error log). 

```suggestion
                }
                
               $this->logger->info(sprintf('Asset: %s removed successfully via event handler', $id));
```

Reply:
No logs are recorded for the happy path of `QtiTestDeleter` anymore.

If assets are to be deleted, `ItemRemovedEventProcessor` records a single message with all the asset URIs before calling `deleteAssetsByURIs`.